templates/features.html

```diff
@@ -74,12 +75,66 @@
   });
 }
 
-document.addEventListener('WebComponentsReady', function(e) {
-  var featureList = $('chromedash-featurelist');
-  var chromeMetadata = $('chromedash-metadata');
-  var overlay = $('chromedash-legend');
-  var search = $('.search input');
+var featureList = $('chromedash-featurelist');
+var chromeMetadata = $('chromedash-metadata');
+var overlay = $('chromedash-legend');
+var search = $('.search input');
+
+// Set search box to URL deep link.
+if (location.hash) {
+  search.value = location.hash.substr(1);
+}
+
+featureList.addEventListener('filter-category', function(e) {
+  search.value = 'category: ' + e.detail.val;
+  updateCount(featureList.filter(search.value));
+});
+
+featureList.addEventListener('filter-owner', function(e) {
+  search.value = 'owner: ' + e.detail.val;
+  updateCount(featureList.filter(search.value));
+});
+
+// Clear input when user clicks the 'x' button.
+search.addEventListener('search', function(e) {
+  if (!e.target.value) {
+    updateCount(featureList.filter());
+    chromeMetadata.selected = null;
+  }
+});
+
+ // Clear input when user clicks the 'x' button.
+search.addEventListener('keyup', function(e) {
+  if (!e.target.value && e.keyCode != 27) { // ESC
+    updateCount(featureList.filter());
+    chromeMetadata.selected = null;
+  } else {
+    //if (e.keyCode == 13) {  // Enter
+      updateCount(featureList.filter(e.target.value));
+    //}
+  }
+});
 
+$('.legend').addEventListener('click', function(e) {
+  e.preventDefault();
+  overlay.toggle();
+});
+
+// Fire of features.json XHR right away so data can populate faster.
+var featuresPromise = new Promise(function(resolve, reject) {
+  var url = location.hostname == 'localhost' ? '/static/js/mockdata.json' :
+                                               '/features.json';
+  var xhr = new XMLHttpRequest(); // Fill features list via AJAX.
+  xhr.open('GET', url);
+  xhr.responseType = 'json';
+  xhr.onloadend = function(e) {
+    resolve(e.target.response);
```

Comment:
A pattern I [like to follow](https://github.com/GoogleChrome/ioweb2015/blob/master/app/scripts/helper/request.js#L115) when Promise-ifying XHR is to check `e.target.status` for anything >= 400, and `reject()` if you get that back, since you'd presumably like to treat HTTP error responses the same way you treat XHR networking failures.